src/solver/expressions/visitors/AstDOTStyleVisitor.cpp

```diff
@@ -0,0 +1,252 @@
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::ParameterNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id, "Param(" + node->value() + ")", NodeStyle::ParameterStyle, os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::LiteralNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id, std::to_string(node->value()), NodeStyle::LiteralStyle, os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::NegationNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id, "-", NodeStyle::NegationStyle, os);
+    int childId = getNodeID(node->child());
+    os << "  " << id << " -> " << childId << ";\n";
+    dispatch(node->child(), os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::PortFieldNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id,
+             "PF(" + node->getPortName() + "," + node->getFieldName() + ")",
+             NodeStyle::PortFieldStyle,
+             os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::ComponentVariableNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id,
+             "CV(" + node->getComponentId() + "," + node->getComponentName() + ")",
+             NodeStyle::ComponentVariableStyle,
+             os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::ComponentParameterNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id,
+             "CP(" + node->getComponentId() + "," + node->getComponentName() + ")",
+             NodeStyle::ComponentParameterStyle,
+             os);
+}
+
+std::string AstDOTStyleVisitor::name() const
+{
+    return "AstGraphVisitor";
+}
+
+unsigned int AstDOTStyleVisitor::getNodeID(const Nodes::Node* node)
+{
+    const auto& node_name = node->name();
+    if (nodeIds_.find(node_name) == nodeIds_.end())
+    {
+        nodeIds_[node->name()][node] = ++nodeCount_;
+    }
+    else if (auto& id_map = nodeIds_[node_name]; id_map.find(node) == id_map.end())
+    {
+        id_map[node] = ++nodeCount_;
+    }
+
+    return nodeIds_[node->name()][node];
+}
+
+void AstDOTStyleVisitor::emitNode(unsigned int id,
+                                  const std::string& label,
+                                  const BoxStyle& box_style,
+                                  std::ostream& os)
+{
+    os << "  " << id << " [label=\"" << label << "\", shape=\"" << box_style.shape << "\", style=\""
+       << box_style.style << "\", color=\"" << box_style.color << "\"];\n";
+}
+
+// Process binary operation nodes like Add, Subtract, etc.
+void AstDOTStyleVisitor::processBinaryOperation(const Nodes::BinaryNode* node,
+                                                const std::string& label,
+                                                const BoxStyle& box_style,
+                                                std::ostream& os)
+{
+    auto id = getNodeID(node);
```

Comment:
Stupid question (sorry I can't find the answer by myself) :
Why do we need **getNodeID** ?
I mean, when visiting the tree, could we not just increment **nodeCount_** and use it to compose the local info to be sent to the stream ?

Reply:
No, a node may not have been identified yet

Reply:
good question! node in the DOT lang must be identified with an unique id ...
why not " just increment nodeCount_ and use it to compose the local info to be sent to the stream" --> because theoretically a node can have multiple parent :

                 A
                / \
               B  C
              /\  /\ 
             .. ..  B

Reply:
ok